cpplint.py

```diff
@@ -50,12 +50,23 @@
 import sre_compile
 import string
 import sys
+import itertools
 import unicodedata
 
+# Files with any of these extensions are considered to be
+# header files (and will undergo different style checks).
+# This set can be extended by using the --headers
+# option (also supported in CPPLINT.cfg)
+_header_extensions = set(['h', 'hpp', 'hxx', 'h++', 'cuh'])
+_nonheader_extensions = set(['c', 'cc', 'cpp', 'cxx', 'c++', 'cu'])
+
+
 # The allowed extensions for file names
 # This is set by --extensions flag.
-_valid_extensions = set(['c', 'cc', 'cpp', 'cxx', 'c++', 'h', 'hpp', 'hxx',
-    'h++'])
+_valid_extensions = _nonheader_extensions.union(_header_extensions)
```

Comment:
I think the confusing and unexpected behaviors already exist in the system as it is, because it uses header extensions throughout the code. So while you are right, this PR does not make matters worse than they are. So this would be for a separate PR, I think